python/GafferUI/ContextQueryUI.py

```diff
@@ -0,0 +1,408 @@
+
+	result = prefix + suffix
+	result = result.replace( ".", "_" ) if sanitise else result
+
+	return result
+
+
+##########################################################################
+# Output widget
+##########################################################################
+
+class _OutputWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, childPlug, **kw ) :
+
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 )
+
+		GafferUI.PlugValueWidget.__init__( self, self.__row, childPlug )
+
+		nameWidget = GafferUI.LabelPlugValueWidget(
+			self.getPlugs(),
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		nameWidget.label()._qtWidget().setFixedWidth( GafferUI.PlugWidget.labelWidth() )
+		self.__row.append(
+			nameWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		existsLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug["exists"] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		existsLabelWidget.label()._qtWidget().setFixedWidth( 40 )
+		self.__row.append(
+			existsLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		self.__row.append(
+			GafferUI.BoolPlugValueWidget( { plug["exists"] for plug in self.getPlugs() } ),
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		valueLabelWidget = GafferUI.LabelPlugValueWidget(
+			{ plug["value"] for plug in self.getPlugs() },
+			horizontalAlignment = GafferUI.Label.HorizontalAlignment.Right
+		)
+		valueLabelWidget.label()._qtWidget().setFixedWidth( 40 )
+		self.__row.append(
+			valueLabelWidget,
+			verticalAlignment = GafferUI.Label.VerticalAlignment.Top
+		)
+
+		self.__row.append(
+			GafferUI.PlugValueWidget.create( { plug["value"] for plug in self.getPlugs() } )
+		)
+
+	def setPlugs( self, plugs ) :
+
+		GafferUI.PlugValueWidget.setPlugs( self, plugs )
+
+		self.__row[0].setPlugs( plugs )
+		self.__row[1].setPlugs( { plug["exists"] for plug in plugs } )
+		self.__row[2].setPlugs( { plug["exists"] for plug in plugs } )
+		self.__row[3].setPlugs( { plug["value"] for plug in plugs } )
+		self.__row[4].setPlugs( { plug["value"] for plug in plugs } )
+
+	def hasLabel( self ) :
+
+		return True
+
+	def childPlugValueWidget( self, childPlug ) :
+
+		for w in self.__row :
+			if childPlug in w.getPlugs() :
+				return w
+
+		return None
+
+	def setReadOnly( self, readOnly ) :
+
+		if readOnly == self.getReadOnly() :
+			return
+
+		GafferUI.PlugValueWidget.setReadOnly( self, readOnly )
+
+		for w in self.__row :
+			w.setReadOnly( readOnly )
+
+
+##########################################################################
+# Metadata
+##########################################################################
+
+Gaffer.Metadata.registerNode(
+
+	Gaffer.ContextQuery,
+
+	"description",
+	"""
+	Queries variables from the current context, creating outputs for each variable.
+	""",
+
```

Comment:
I think we probably want to style the ContextQuery like expression, animation, random and spreadsheet - see https://github.com/GafferHQ/gaffer/blob/main/python/GafferUI/ExpressionUI.py#L56-L61 for relevant metadata.

Reply:
Turning off `resizeWhenInputsChange` for the ArrayPlug has caused the appearance of an unwanted `+` button when using the StandardNodeGadget :

![image](https://user-images.githubusercontent.com/1133871/172198109-27b3a10a-45a8-4c2f-80f5-77236d4ac171.png)

Ideally I guess we'd have a custom `+` button that called `addQuery()` appropriately, but I think the NodeEditor UI for that is sufficient for now. We can disable the unwanted button by registering this metadata against the `out` plug :

```"noduleLayout:customGadget:addButton:gadgetType", ""```